codec_packetizers/h265/h265_depacketizer.c

```diff
@@ -226,14 +242,17 @@ H265Result_t H265Depacketizer_AddPacket( H265DepacketizerContext_t * pCtx,
     H265Result_t result = H265_RESULT_OK;
 
     if( ( pCtx == NULL ) ||
-        ( pPacket == NULL ) )
+        ( pPacket == NULL ) ||
+        ( pPacket->pPacketData == NULL ) ||
+        ( pPacket->packetDataLength == 0 ) )
     {
         result = H265_RESULT_BAD_PARAM;
     }
 
     if( result == H265_RESULT_OK )
     {
-        if( pCtx->packetCount >= pCtx->packetsArrayLength )
+        if( ( pCtx->packetCount >= pCtx->packetsArrayLength ) ||
+            ( pCtx->headIndex >= pCtx->packetsArrayLength ) )
```

Comment:
likely headIndex and packetCount are duplicated. Can we just remove one of them?